src/main/java/org/coffeemine/app/spring/components/AddProjectDialog.java

```diff
@@ -4,31 +4,64 @@
 import com.vaadin.flow.component.button.Button;
 import com.vaadin.flow.component.button.ButtonVariant;
 import com.vaadin.flow.component.dialog.Dialog;
-import com.vaadin.flow.component.formlayout.FormLayout;
-import com.vaadin.flow.component.html.H1;
+import com.vaadin.flow.component.html.H2;
+import com.vaadin.flow.component.html.Span;
+import com.vaadin.flow.component.notification.Notification;
 import com.vaadin.flow.component.orderedlayout.VerticalLayout;
 import com.vaadin.flow.component.textfield.TextField;
+import com.vaadin.flow.data.value.ValueChangeMode;
+
+import org.coffeemine.app.spring.data.Project;
+import org.coffeemine.app.spring.db.NitriteDBProvider;
 
 public class AddProjectDialog extends Dialog {
 
     public AddProjectDialog() {
         VerticalLayout layout = new VerticalLayout();
         this.add(layout);
 
-        FormLayout columnLayout = new FormLayout();
-        columnLayout.getStyle().set("padding", "15px");
-        TextField projectName = new TextField();
-        projectName.setPlaceholder("Project name");
+        final var columnLayout = new VerticalLayout();
+        columnLayout.getStyle().set("padding", "0px");
+
+        final var addCleanButton = new Button("Create");
+        addCleanButton.getStyle().set("margin", "16px 0px 0px auto");
+        addCleanButton.addThemeVariants(ButtonVariant.MATERIAL_CONTAINED);
+        addCleanButton.setEnabled(false);
+
+        final var projectName = new TextField("Project name");
+        projectName.setWidthFull();
+        projectName.getStyle().set("margin", "0px");
+        projectName.setRequired(true);
+        projectName.setValueChangeMode(ValueChangeMode.EAGER);
+
+        projectName.addValueChangeListener(text -> {
+            if(text.getValue().length() > 0) {
+                addCleanButton.setEnabled(true);
+            } else {
+                addCleanButton.setEnabled(false);
+            }
+        });
+
+        final var projectDescription = new TextField("Description");
+        projectDescription.setWidthFull();
+        projectDescription.getStyle().set("margin", "0px");
+
+        addCleanButton.addClickListener(c -> {
+            try {
+                final var newProjectName = projectName.getOptionalValue().orElseThrow();
+                NitriteDBProvider.getInstance().addProject(new Project(newProjectName, 0));
+                Notification.show("Project \"" + newProjectName+ "\" created", 2000, Notification.Position.BOTTOM_END);
+                close();
+            } catch (Exception e) {}
```

Comment:
Exceptions for control flow is no thanks
I love it for the memes, but it's definitely a bad idea in all languages where stack unwinding is heavy

Reply:
hmm, my thinking was that this avoids a bunch of if statements